src/rules/no-useless-path-segments.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * @fileOverview Ensures that there are no useless path segments
+ * @author Thomas Grainger
+ */
+
+import path from 'path'
+import sumBy from 'lodash.sumby'
+import resolve from 'eslint-module-utils/resolve'
+import moduleVisitor from 'eslint-module-utils/moduleVisitor'
+
+/**
+ * convert a potentialy relative path from node utils into a true
+ * relative path.
+ *
+ * ../ -> ..
+ * ./ -> .
+ * .foo/bar -> ./.foo/bar
+ * ..foo/bar -> ./..foo/bar
+ * foo/bar -> ./foo/bar
+ *
+ * @param rel {string} relative posix path potentially missing leading './'
+ * @returns {string} relative posix path that always starts with a ./
+ **/
+function toRel(rel) {
+  const stripped = rel.replace(/\/$/g, '')
+  return stripped.match(/^((\.\.)|(\.))($|\/)/) ?
```

Comment:
since we're not using the match object here, let's use `regex.test(str)` instead.